### PR TITLE
bpf: Refactoring egress gateway datapath

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request performs just a bit of refactoring on the egress gateway's code in the datapath. This refactoring was prepared for the SRv6 work (cf. https://github.com/cilium/cilium/issues/17646) but is independent so sending it's own pull request.

See commits for details.

### CODE REVIEW

**Review by borkmann — APPROVED** [review]
lgtm

File: bpf/lib/egress_policies.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2021 Authors of Cilium */
+
+#ifndef __LIB_EGRESS_POLICIES_H_
+#define __LIB_EGRESS_POLICIES_H_
+
+#ifdef ENABLE_EGRESS_GATEWAY
+/* is_egress_destination returns true if the given destination is part of the
```
Comment (kkourt) [inline bpf/lib/egress_policies.h]: Seems like the function name was renamed, but not the comment.

```suggestion
/* is_cluster_destination returns true if the given destination is part of the
```

nit: would `is_cluster_local` be a better name?

File: bpf/lib/nodeport.h [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include "lb.h"
 #include "common.h"
 #include "overloadable.h"
+#include "egress_policies.h"
```
Comment (kkourt) [inline bpf/lib/nodeport.h:16]: nit: are there any policies that are not egress gw policies, that could be characterized as egress policies?

**Review by kkourt — APPROVED** [review]
LGTM, thanks.

File: bpf/lib/nodeport.h (hunk 2 above)
Comment (pchaigno) [reply bpf/lib/nodeport.h:16]: SRv6 egress policies are coming :smiley:

File: bpf/lib/egress_policies.h (hunk 1 above)
Comment (pchaigno) [reply bpf/lib/egress_policies.h]: Nice catch, thanks!

**Comment by pchaigno** [comment]
As seen in https://github.com/cilium/cilium/compare/ca13a74c4d8fe856d34c3892df528215bc927cd6..ef858bbab4d78bdc111d35f73947a9c8d6dbdda6, the only change of the last push is a comment change. All tests were green before that, so not triggering again and keeping ready-to-merge label.

**Comment by pchaigno** [comment]
Added label for backporting to v1.10 to facilitate the backport of other egress gateway fixes.

**Pull request merged**